lib/pliny/librato/metrics/backend.rb

```diff
@@ -54,21 +57,13 @@ def start_timer
           end
         end
 
-        def start_counter
-          @counter = Thread.new do
-            loop do
-              msg = metrics_queue.pop
-              msg == POISON_PILL ? break : enqueue_librato(msg)
-            end
-          end
-        end
-
-        def enqueue_librato(msg)
-          sync { librato_queue.add(msg) }
-        end
-
         def flush_librato
-          sync { librato_queue.submit }
+          sync do
+            counter_cache.flush_to(librato_queue)
+            librato_queue.merge!(aggregator)
+            aggregator.clear
+          end
+          librato_queue.submit
```

Comment:
@gudmundur thanks for pointing this out... I'm assuming the "poison pill" approach that was used previously is how you avoided race conditions and blocking?